src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -162,6 +163,11 @@ public override void Load()
 
         public async Task RefreshAsync(CancellationToken cancellationToken)
         {
+            if (!_configClientManager.HasAvailableClients)
+            {
+                return;

```

Comment:
Not adding any logs now, we can add success logs in the future.

Reply:
If we move the availability client check after the watcher expiration check, the log won't be too crazy. If we are concerned, we can do a debug level log.

Reply:
> If we move the availability client check after the watcher expiration check, the log won't be too crazy

That depends on the state of the watchers. If we're in backoff, and the watchers are due for refresh, then I expect the situation to be the same. Also, I would expect that if we're in backoff, it's most likely because the watchers are due for refresh.

Reply:
Let's settle on the availability check in my the other comment. Then come back for the logging.

Reply:
Let's handle logging in a separate PR with telemetry.